packages/strapi-plugin-content-type-builder/services/schema-builder/schema-handler.js

```diff
@@ -76,6 +76,15 @@ module.exports = function createSchemaHandler(infos) {
       return this;
     },
 
+    // Set a particular option inside the schema
+    setSchemaOption(path, val) {
+      modified = true;
+
+      _.set(state.schema, ['options', ...path], val);
```

Comment:
you can do ['options'].concat(path) so we can call setOption('draftAndPublish', value)